Support IDENTITY() function in SELECT-INTO

deepakshi-mittal opened the pull request:
### Description
BABEL-539

Engine PR: https://github.com/babelfish-for-postgresql/postgresql_modified_for_babelfish/pull/164/commits

[Describe what this change achieves - Guidelines below (please delete the guidelines after writing the PR description)]

> 1. *What* is the change? This is best described in terms of “Currently, Babelfish does X. With this change it now does Y.” Think of “What *did* it *used* to do?” and “What *does* it do *now*?”
2. *Why* was the change made? What drove our desire to put effort into the change?
3. *How* was the code changed should only appear for large commits. This can serve as a rough roadmap to what’s contained in the commit. It should be very high level; if it’s directly referencing code it’s probably too detailed. It’s also critical that this section of a commit message does not try to replace proper code documentation (ie, block comments or README files). Generally, this section should only appear if the commit itself is large enough that it’s helpful to provide a roadmap to someone looking at the commit.
4. The last descriptive piece is the “title” for the commit: the very first line of the commit message, which should typically be less than 80 characters. A good title is *critical*, because it’s the only thing that shows up in places like the Github commit listing. No one’s got time to read through full commit messages when trying to find a single commit out of dozens.


### Issues Resolved

[List any issues this PR will resolve]

### Test Scenarios Covered ###
* **Use case based -**


* **Boundary conditions -**


* **Arbitrary inputs -**


* **Negative test cases -**


* **Minor version upgrade tests -**


* **Major version upgrade tests -**


* **Performance tests -**


* **Tooling impact -**


* **Client tests -**



### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

deepakshi-mittal committed:
Support IDENTITY() function in SELECT-INTO

Task: BABEL-539

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Identity in Select Into Added Test case

Task: BABEL-539

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Removed extra code,added formatting

Task: BABEL-539

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Empty Commit

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

forestkeeper commented on contrib/babelfishpg_tsql/runtime/functions.c:
```diff
@@ -1844,6 +1845,20 @@ language(PG_FUNCTION_ARGS)
+		result = nextval_internal(tsql_select_into_seq_oid, false);
+	}
+	else{
+		result = 0;
```
We should throw an runtime error here or add a Assert(false) here, since I think execution would never be here.

forestkeeper commented on contrib/babelfishpg_tsql/src/pl_handler.c:
```diff
@@ -171,6 +173,7 @@ static Oid	lang_handler_oid = InvalidOid;	/* Oid of language handler
 											 * function */
 static Oid	lang_validator_oid = InvalidOid;	/* Oid of language validator
 												 * function */
+Oid tsql_select_into_seq_oid = InvalidOid;
```
Please add a comment explain why we have this variable

deepakshi-mittal committed:
Remaned SelectIntoutility hook

Added Assert for Oid
Added function in upgrade script

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
temporary add global variable for debug Typename

Task: BABEL-539

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Commdnted unused variable

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Added Hook for handling of identity column before creating relation

Added Hook for create relation
Changed function parameter for identity_into

Task: BABEL-539

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Corrected test cases for upgrade failure

Task: BABEL-539

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal committed:
Moved function up to resolve merge conflict

Signed-off-by: Deepakshi Mittal <depakshi@amazon.com>

deepakshi-mittal commented:
Has conflicts , creating new branch
